lib/config/validation.ts

```diff
@@ -603,6 +603,22 @@ export async function validateConfig(
                   message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
                 });
               }
+            } else if (key === 'env') {
+              const allowedEnvVars = GlobalConfig.get('allowedEnv')!;
```

Comment:
Does it fail when we've not set global config yet? eg when we validate the global config?

Reply:
Yep. This is also true for all options which depend on a GlobalConfig.get() ... will need to fix that first.
I guess depending upon globalConfig: true or false ... we should switch between using the value directly from the config or the GlobalConfig object.

ie. 
```ts
const allowedEnvVars = isGlobalConfig ? config.allowedEnv : GlobalConfig.get('allowedEnv')!;
```

Reply:
Created one here: https://github.com/renovatebot/renovate/pull/27261

Reply:
We still have `customEnvVariables` so I was planning that this `env` option was for repo config only. Does that change anything?

Reply:
Nope, env is repo only from the beginning. Am I missing something?

Reply:
`env` relies on `allowedEnv` which is global only. So the validation won't change much.

Reply:
I thought it changes things because then @viceice's question here is no longer applicable

Reply:
Got it. Since its not intended to reside in file config we do not need to worry about validation fails there.